py_trees/decorators.py

```diff
@@ -0,0 +1,457 @@
+#!/usr/bin/env python
+#
+# License: BSD
+#   https://raw.githubusercontent.com/stonier/py_trees/devel/LICENSE
+#
+##############################################################################
+# Documentation
+##############################################################################
+
+"""
+Decorators are behaviours that manage a single child and provide common
+modifications to their underlying child behaviour (e.g. inverting the result).
+i.e. they provide a means for behaviours to wear different 'hats' depending
+on their context without a behaviour tree.
+
+.. image:: images/many-hats.png
+   :width: 40px
+   :align: center
+
+An example:
+
+.. graphviz:: dot/decorators.dot
+
+.. literalinclude:: examples/decorators.py
+   :language: python
+   :linenos:
+
+
+**Decorators (Hats)**
+
+Decorators with very specific functionality:
+
+* :func:`py_trees.decorators.Condition`
+* :func:`py_trees.decorators.Inverter`
+* :func:`py_trees.decorators.OneShot`
+* :func:`py_trees.decorators.TimeOut`
+
+And the X is Y family:
+
+* :func:`py_trees.decorators.Failure_is_running`
+* :func:`py_trees.decorators.Failure_is_success`
```

Comment:
typo: *FailureIsRunning and *FailureIsSuccess

Reply:
Done